### DISCUSSION TITLE
[feat] webhid API

### DISCUSSION THREAD

**Discussion by HaoboGu** [tauri-apps/tauri#14550] [Plugin Requests]
### Describe the problem

Electron has APIs for device accessing, like webhid/webusb/webserial. I cannot find any equivalent in tauri. Electron also provides a way to skip the popup to grant access of the device.

reference: https://www.electronjs.org/docs/latest/tutorial/devices#webhid-api

### Describe the solution you'd like

Add same APIs

**Comment by k26pl** [comment]
These are chromium-only apis, and are not supported on other webwiev engines which are used by tauri. A possible workaround would be using rust usb library.

**Comment by HigherOrderLogic** [comment]
Hi, I'm also interest in getting this feature implemented.

Can't it be guarded behind some kind if feature flag, and exit the build process if the webview engine isn't supported?

**Comment by kingraphaii** [comment]
+1 

@k26pl when you say:

> These are chromium-only apis, and are not supported on other webwiev engines which are used by tauri. A possible workaround would be using rust usb library.

Is there another way of doing this if the HID isn't USD connected? Like for example an integrated laser scanner.

**Comment by Jaquinei** [comment]
The **hid** object is exposed by the navigator object in the Tauri WebView.

<img width="376" height="282" alt="Image" src="https://github.com/user-attachments/assets/14a8f413-4321-40b8-b948-f2acf4526d28" />

Is HID support already available in the latest WebView2 version in Tauri on Windows?

**Comment by FabianLars** [comment]
yes, and probably always was since the windows webview is based on chromium. this issue really is about the other platforms, that just wasn't specified.

**Comment by Jaquinei** [comment]
> yes, and probably always was since the windows webview is based on chromium. this issue really is about the other platforms, that just wasn't specified.

Thanks for the information.
I’d just like to double-check: on Windows, WebView2 (which is Chromium-based) does not have any unexpected restrictions regarding the WebHID API, correct?
On macOS, WebHID is not supported in WKWebView, as Apple has chosen not to implement the WebHID API in either Safari or their WebView component.

**Comment by FabianLars** [comment]
> I’d just like to double-check: on Windows, WebView2 (which is Chromium-based) does not have any unexpected restrictions regarding the WebHID API, correct?

I'm not aware of any restrictions compared to any other chromium based browser.

> On macOS, WebHID is not supported in WKWebView, as Apple has chosen not to implement the WebHID API in either Safari or their WebView component.

Yep, same for Firefox btw https://mozilla.github.io/standards-positions/#webhid

**Comment by ezra-en** [comment]
Correct me if I'm wrong but it sounds like we might have a solution in https://github.com/RedfernElec/tauri-plugin-hid ?
